integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/glue/steps/database/device/DlmsDeviceSteps.java

```diff
@@ -503,28 +512,31 @@ public void aValidMbusUserKeyIsStored(final Map<String, String> settings) {
     final List<DbEncryptedSecret> securityKeys = this.findAllSecretsForDevice(deviceIdentification);
 
     int numberOfMbusDefaultKeys = 0;
-    int numberOfMbusUserKeys = 0;
-    int numberOfValidMbusUserKeys = 0;
+    int numberOfMbusKeys = 0;
+    int numberOfActiveMbusKeys = 0;
 
     for (final DbEncryptedSecret securityKey : securityKeys) {
-      switch (securityKey.getSecretType()) {
-        case G_METER_MASTER_KEY:
-          numberOfMbusDefaultKeys += 1;
-          break;
-        case G_METER_ENCRYPTION_KEY:
-          numberOfMbusUserKeys += 1;
-          if (securityKey.getSecretStatus().equals(SecretStatus.ACTIVE)) {
-            numberOfValidMbusUserKeys += 1;
-          }
-          break;
-        default:
-          // other keys are not counted
+      if (securityKey.getSecretType().equals(G_METER_MASTER_KEY)) {
+        numberOfMbusDefaultKeys += 1;
+      } else if (securityKey.getSecretType().equals(secretType)) {
+        numberOfMbusKeys += 1;
+        if (securityKey.getSecretStatus().equals(SecretStatus.ACTIVE)) {
+          numberOfActiveMbusKeys += 1;
+        }
       }
     }
 
     assertThat(numberOfMbusDefaultKeys).as("Number of M-Bus Default keys stored").isEqualTo(1);
-    assertThat(numberOfMbusUserKeys > 0).as("At least one M-Bus User key must be stored").isTrue();
-    assertThat(numberOfValidMbusUserKeys).as("Number of valid M-Bus User keys stored").isEqualTo(1);
+    assertThat(numberOfMbusKeys >= storedKeyCount)
+        .as(
+            "At least " + storedKeyCount + " M-Bus key of the specified type must be stored",
+            storedKeyCount)
+        .isTrue();
```

Comment:
assertThat(numberOfMbusKeys).**isGreaterThanOrEqualTo**(storedKeyCount).as(...)

Reply:
assert is updated.